akka-http-core/src/main/scala/akka/http/scaladsl/ClientTransport.scala

```diff
@@ -57,11 +56,15 @@ object ClientTransport {
    */
   def httpsProxy(proxyAddress: InetSocketAddress): ClientTransport = new HttpsProxyTransport(proxyAddress)
 
-  private case class HttpsProxyTransport(proxyAddress: InetSocketAddress, underlyingTransport: ClientTransport = TCP) extends ClientTransport {
+  def httpsProxy(proxyAddress: InetSocketAddress, proxyAuth: HttpCredentials): ClientTransport = new HttpsProxyTransport(proxyAddress, proxyAuth = Some(proxyAuth))
```

Comment:
Not a big scaladoc guru. Should I just be copy/paste the scala doc above and adding info about the optional proxy? 

Also, the ClientTransport object already has the @ApiMayChange marker, so do I also need to add it on the method?